### ISSUE TITLE
Translate Mango Python integration tests to Elixir

### ISSUE THREAD

**Issue by pgj** [apache/couchdb#4501]
Currently, Mango has its own integration tests that are implemented in Python and have to be activated separately via the `mango-tests` `make(1)` target.  Re-write them in Elixir to integrate them better with the rest of the integration tests that are in use for CouchDB.  This move could bring more simplification and regularization to the code base, removing the dependency on Nose2.

**Comment by nickva** [comment]
Good idea. It could be done step-wise, one file or feature at a time. Move from one test suite to the other over time. We did it similarly with Javascript -> Elixir move a few years back.

If bringing in Nose2 is a hassle, another way could be to try to see if Python 3's built-in https://docs.python.org/3/library/unittest.html#module-unittest has the feature nose2 uses. The tests were written during the Python2 days and Python had some improvements since.

**Comment by pgj** [comment]
Personally, I do not have any problems with Nose2.  But if the integration tests were migrated to Elixir, there is no need for it any more.  On its web site, it considers itself obsolete -- but instead of moving to another Python-based framework, consider Elixir directly.

**Comment by big-r81** [comment]
Personally, I wouldn't write (or migrate) any test to Elixir, because I see no benefits in learning and using a second/third programming language to add contributions to CouchDB. As a new developer to CouchDB (and Erlang), I found using a second language more of an obstacle than an asset.

**Comment by pgj** [comment]
I am not sure if I got you right @big-r81.  Elixir has already been used in CouchDB and there are a lot of integration tests written in that language.  And, on the top of that, there is Python for Mango.

Or are you saying that we should just leave Python there, because the chances are likely that new contributors are already familiar with it?

**Comment by big-r81** [comment]
@pgj yes, there are a lot of Elixir tests (from the migration from Javascript, if I get this right). This is only a personal preference of me, not to use Elixir for tests. I think the move to using Elixir was made back then to attract more developers. I don't think this happened...
But feel free to use Elixir. 🙂

**Comment by jjrodrig** [comment]
From my experience as an Elixir test suite contributor, I found that Elixir allows new contributions much easier than Erlang does. 

In this case, Python is the exception in the test suite, so I think it is a good idea to simplify this part and migrate these tests to the integration test suite based on Elixir.
